marshmallow/marshalling.py

```diff
@@ -181,6 +183,7 @@ def run_validator(self, validator_func, output,
                 raise ValidationError(self.default_schema_validation_error)
         except ValidationError as err:
             errors = self.get_errors(index=index)
+            self.kwargs = err.kwargs
```

Comment:
Since multiple `ValidationErrors` may be raised during validation, it might make sense to do `self.kwargs.update(err.kwargs)`.